File: apps/desktop/src-tauri/src/commands/system.rs
use super::ErrorPayload;
use crate::store;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ApiEnvDiagnostic {
    pub openai_api_key_set: bool,
    pub gemini_api_key_set: bool,
    pub anthropic_api_key_set: bool,
}

#[tauri::command]
pub fn db_status() -> store::DbStatus {
    store::db_status()
}

#[tauri::command]
pub fn get_system_agents() -> Result<repodesk_core::agents::AgentsConfig, ErrorPayload> {
    repodesk_core::agents::ensure_agents_config().map_err(ErrorPayload::from)
}

#[tauri::command]
pub fn get_system_capabilities()
-> Result<repodesk_core::capabilities::CapabilitiesConfig, ErrorPayload> {
    repodesk_core::capabilities::ensure_capabilities_config().map_err(ErrorPayload::from)
}

#[tauri::command]
pub fn get_system_peripherals()
-> Result<repodesk_core::peripherals::PeripheralsConfig, ErrorPayload> {
    repodesk_core::peripherals::ensure_peripherals_config().map_err(ErrorPayload::from)
}

#[tauri::command]
pub fn get_system_modules() -> Vec<repodesk_core::module_registry::BrainModule> {
    repodesk_core::module_registry::list_modules()
}

File: apps/desktop/src/shared/api/system.ts
import { invoke } from "@tauri-apps/api/core";

export type DbStatus = {
  path: string;
  exists: boolean;
  ok: boolean;
  tables: string[];
  error?: string | null;
};

export type ApiEnvDiagnostic = {
  openai_api_key_set: boolean;
  gemini_api_key_set: boolean;
  anthropic_api_key_set: boolean;
};

export type AgentConfig = {
  name: string;
  kind: string;
  role: string;
  default_budget_tokens: number;
  allowed_actions: string[];
  forbidden_actions: string[];
  preferred_for: string[];
};

export type AgentsConfig = {
  agents: AgentConfig[];
};

export type Capability = {
  name: string;
  kind: string;
  enabled: boolean;
  local: boolean;
  risk: string;
  boundary: string;
  preferred_for: string[];
  allowed_actions: string[];
  forbidden_actions: string[];
};

export type CapabilitiesConfig = {
  capabilities: Capability[];
};

export type PeripheralConfig = {
  name: string;
  kind: string;
  access: string;
  risk: string;
  allowed_actions: string[];
  forbidden_actions: string[];
};

export type PeripheralsConfig = {
  peripherals: PeripheralConfig[];
};

export type BrainModule = {
  name: string;
  layer: string;
  status: string;
  purpose: string;
};

export async function dbStatus(): Promise<DbStatus> {
  return invoke("db_status");
}

export async function getApiEnvDiagnostic(): Promise<ApiEnvDiagnostic> {
  return invoke("get_api_env_diagnostic");
}

export async function getSystemAgents(): Promise<AgentsConfig> {
  return invoke("get_system_agents");
}

export async function getSystemCapabilities(): Promise<CapabilitiesConfig> {
  return invoke("get_system_capabilities");
}

export async function getSystemPeripherals(): Promise<PeripheralsConfig> {
  return invoke("get_system_peripherals");
}

export async function getSystemModules(): Promise<BrainModule[]> {
  return invoke("get_system_modules");
}
